dask/dataframe/utils.py

```diff
@@ -443,6 +444,8 @@ def _nonempty_series(s, idx=None):
             cats = None
         data = pd.Categorical(data, categories=cats,
                               ordered=s.cat.ordered)
+    elif type(dtype) in make_array_nonempty._lookup:
```

Comment:
It might be slightly cleaner to implement `_nonempty_series` using `make_array_nonempty`. I haven't attempted that yet.